src/components/ProjectSettingsTabs/CreateProjectModal.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Size, Variant } from "@leafygreen-ui/button";
+import { Button } from "components/Button";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm, SpruceFormProps } from "components/SpruceForm";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  CreateProjectMutation,
+  CreateProjectMutationVariables,
+} from "gql/generated/types";
+import { CREATE_PROJECT } from "gql/mutations";
+
+const getModalFormDefinition = (owner: string, repo: string) => ({
+  schema: {
+    type: "object" as "object",
+    properties: {
+      projectName: {
+        type: "string" as "string",
+        title: "Project Name",
+      },
+      projectId: {
+        type: "string" as "string",
+        title: "Project Id",
+        value: "hi",
+      },
+      owner: {
+        type: "string" as "string",
+        title: "Owner",
+        default: owner,
+      },
+      repo: {
+        type: "string" as "string",
+        title: "Repo",
+        default: repo,
+      },
+    },
+  },
+  uiSchema: {
+    projectName: {
+      "ui:data-cy": "project-name-input",
+      "ui:showErrors": false,
+    },
+    projectId: {
+      "ui:data-cy": "project-id-input",
+      "ui:description":
+        "Optionally enter an immutable project ID that would be used by Evergreen internally instead of defaulting to a random hash; An id should only be user-specified with good reason, such as if the project will be using performance tooling. \n It cannot be changed!",
```

Comment:
I'm still looking into a way to get newlines working, so may as well remove for now.
```suggestion
        "Optionally enter an immutable project ID that would be used by Evergreen internally instead of defaulting to a random hash; An ID should only be user-specified with good reason, such as if the project will be using performance tooling. It cannot be changed!",
```